addons/web/static/src/core/signature/name_and_signature.js

```diff
@@ -245,14 +245,22 @@ export class NameAndSignature extends Component {
      */
     async printImage(imgSrc) {
         this.clear();
-        // Force the width and height to the canvas size to prevent shrinking
-        // high DPI screens (devicePixelRatio > 1)
-        const options = {
-            width: this.signatureRef.el.width,
-            height: this.signatureRef.el.height,
-            yOffset: 30,
+        const c = this.signaturePad.canvas;
+        const img = new Image();
+        img.onload = () => {
+            const ctx = c.getContext("2d");
+            var ratio = ((img.width / img.height) > (c.width / c.height)) ? c.width / img.width : c.height / img.height;
+            ctx.drawImage( 
+                img,
+                (c.width / 2) - (img.width * ratio / 2),
+                (c.height / 2) - (img.height * ratio / 2)
+                , img.width * ratio
+                , img.height * ratio
+            );
+            this.props.signature.isSignatureEmpty = this.isSignatureEmpty;
         };
-        await this.fromDataURL(imgSrc, options);
```

Comment:
No, we don't need it anymore.

The problem with `fromDataURL` is that we cannot call it with the right `options` without loading the image before calling it. but it will load the image again anyway. so I just removed it, loaded the image only once and used the `signaturePad` canvas directly to draw the image.